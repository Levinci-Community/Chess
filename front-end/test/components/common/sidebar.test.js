import React from 'react';
import renderer from 'react-test-renderer';
import Sidebar from '../../../src/components/common/sidebar';
import { BrowserRouter as Router } from 'react-router-dom';
import { ChakraProvider } from "@chakra-ui/react";

const data = [
  { id: 1, title: 'Home', link: '/' },
  { id: 2, title: 'About', link: '/about' },
];

describe('Components > Sidebar', () => {
  it('renders a snapshot for Sidebar using renderer', () => {
    const tree = renderer.create(
      <ChakraProvider>
        <Router>
          <Sidebar data={data} isOpen={true} onClose={() => {}} />
        </Router>
      </ChakraProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
