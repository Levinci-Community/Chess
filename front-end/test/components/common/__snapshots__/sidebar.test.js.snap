// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Sidebar renders a snapshot for Sidebar using renderer 1`] = `
[
  <span />,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
